### PR TITLE
Develop to master

### PR DESCRIPTION
**Pull request by Eptis**

### CODE REVIEW

File: lib/helpers/filter.ex [hunk 1]
```diff
@@ -0,0 +1,24 @@
+defmodule RogerUi.Helpers.Filter do
+  @moduledoc """
+  Take a enumeration and returns only filtered element
+  """
+
+  @spec call(
+          xs :: Enumerable.t(),
```
Comment (Eptis) [inline lib/helpers/filter.ex:7]: What does `xs` mean? Maybe rename this variable.

File: lib/helpers/page.ex [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  """
+
+  @doc """
+  Slice enumerable into pages of a given page_size and returns a map with
+  a list, which represents the page of enumerable in the given page_number
+  and the total elements of the enumerable.
+  Max page_size is 100 and min page_number is 1
+  """
+  @spec extract(
+          xs :: Enumerable.t(),
```
Comment (Eptis) [inline lib/helpers/page.ex]: What does `xs` mean? Maybe rename this variable.

File: lib/helpers/response.ex [hunk 3]
```diff
@@ -0,0 +1,30 @@
+defmodule RogerUi.Helpers.Response do
+  @moduledoc """
+  CORS and JSON
+  """
+  import Plug.Conn
+
+  defp set_cors_headers(conn) do
```
Comment (Eptis) [inline lib/helpers/response.ex]: Can you place private functions at the bottom? I think it makes sense to first display the public API first and then the implementation.

File: lib/roger_ui/jobs.ex [hunk 4]
```diff
@@ -0,0 +1,21 @@
+defmodule RogerUi.Jobs do
+  @moduledoc """
+  Generate Jobs list from Roger.Info.running_jobs()
+  """
+
+  defp normalize_jobs({partition_name, jobs}) do
+    Stream.map(jobs, fn j ->
+      j
```
Comment (Eptis) [inline lib/roger_ui/jobs.ex]: Can you rename `j` to `job`?

File: lib/roger_ui/roger.ex [hunk 5]
```diff
@@ -0,0 +1,38 @@
+defmodule RogerUi.Roger do
+  @moduledoc """
+  Contract to Roger calls
+  """
+
+  @type queue :: %{
```
Comment (Eptis) [inline lib/roger_ui/roger.ex]: Can you add `@typedoc` for the public types or make them private (`@typep`) if they should not be referenced outside this module?

File: priv/vue/src/components/Jobs.vue [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    },
+
+    update_checked (checkedStatus) {
+      this.checked = checkedStatus.checked
+      this.nothing_selected = checkedStatus.nothing_selected
+      this.all_selected = checkedStatus.all_selected
+    },
+
+    cancel () {
+      if (this.nothing_selected) return
```
Comment (Eptis) [inline priv/vue/src/components/Jobs.vue:69]: shouldn't this return a completed promise to make `cancel()` safe to chain?

File: lib/helpers/filter.ex (hunk 1 above)
Comment (Spadavecchia) [reply lib/helpers/filter.ex:7]: changed to `enumerable`

File: lib/helpers/page.ex (hunk 2 above)
Comment (Spadavecchia) [reply lib/helpers/page.ex]: changed to `enumerable`

File: lib/helpers/response.ex (hunk 3 above)
Comment (Spadavecchia) [reply lib/helpers/response.ex]: All private functions moved to end of module

File: lib/roger_ui/jobs.ex (hunk 4 above)
Comment (Spadavecchia) [reply lib/roger_ui/jobs.ex]: Renamed j -> job

File: lib/roger_ui/roger.ex (hunk 5 above)
Comment (Spadavecchia) [reply lib/roger_ui/roger.ex]: types declared privated

File: priv/vue/src/components/Jobs.vue (hunk 6 above)
Comment (Spadavecchia) [reply priv/vue/src/components/Jobs.vue:69]: the function is `impure` and returning `void` because exists only to mutate the state of jobs array

**Comment by Spadavecchia** [comment]
Implemented in the lastest commits to `develop` branch

**Pull request closed without merging**